rest-client/src/main/java/org/sagebionetworks/bridge/rest/AuthenticationHandler.java

```
@@ -54,7 +56,11 @@ public AuthenticationHandler(SignIn signIn, UserSessionInfoProvider userSessionI
         }
         tryCount++;
 
-        userSession = userSessionInfoProvider.retrieveSession(signIn);
+        try {
+            userSession = userSessionInfoProvider.retrieveSession(signIn);
+        } catch (ConsentRequiredException e) {
+            LOG.debug("Ignoring 412 during auto-authentication");
```

I think you should just let this throw. It would be unexpected and unusual, and would indicate that further calls are going to fail...

---

I thought this might be the case also, until I saw what would happen to this line:
`user.getClient(ParticipantsApi.class).getUsersParticipantRecord().execute();`

If the user's session is current, the call succeeds.

If the user's session is invalid (expired, or logged out), the call would throw a 412, even though the API being called does not require consent. The call would fail due to something that we're aiming to encapsulate (auto-reauth in case of session expiry), so throwing a 412 sort of an unexpected side effect. Running the exact same line again would succeed.

I'm leaning towards hiding this exception. Unless there is a use case where we'd want to know the participant has not consented, when we're calling an API that doesn't require consent (if it did, a 412 would get thrown anyway, and the stack trace would show it coming from the API being called, instead of an authenticator being called internally by OkHttp)

---

You don't have to be consented to get the user's own participant record. Signing in at that point, if you were not signed in and not consented, it would return 412 (unless you have administrative roles, then it wouldn't). 

You should never hide this exception: for users with roles using administrative APIs, it isn't thrown (even on sign in), and for APIs where it is returned, they will never work until the user signs the consent. It should be thrown out of our core libraries... applications when they get it, should go into the consent system.

Another problem with hiding this is if we update the application, introducing a new consent, and signing everyone out, a call intercepted by this service would not do the right thing, and send the user to consent, if this is swallowed.

---

Okay, you're right, a 412 should interrupt whatever the user or background thread is doing, until consent is obtained. Thanks, I'll make the change!